agenthub/langchains_agent/utils/monologue.py

```diff
@@ -7,17 +9,29 @@ def __init__(self, model_name):
         self.model_name = model_name
 
     def add_event(self, t: dict):
-        self.thoughts.append(t)
+        # Validate that the event is a dictionary
+        if not isinstance(t, dict):
+            raise ValueError("Event must be a dictionary")
+        # Add a timestamp to each event for metadata
+        event_with_metadata = t.copy()
+        event_with_metadata["timestamp"] = datetime.datetime.now().isoformat()
+        self.thoughts.append(event_with_metadata)
 
     def get_thoughts(self):
         return self.thoughts
 
     def get_total_length(self):
-        return sum([len(json.dumps(t)) for t in self.thoughts])
+        total_length = 0
+        for t in self.thoughts:
+            try:
+                total_length += len(json.dumps(t))
+            except TypeError as e:
+                print(f"Error serializing thought: {e}")
+        return total_length
 
     def condense(self):
-        new_thoughts = llm.summarize_monologue(self.thoughts, self.model_name)
-        # self.thoughts = [Event(t['action'], t['args']) for t in new_thoughts]
-        self.thoughts = new_thoughts
-
-
+        try:
```

Comment:
I think this should be a crash--otherwise the context window could expand ad infinitum